pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -897,43 +923,44 @@ func (pl *dynamicResources) Reserve(ctx context.Context, cs *framework.CycleStat
 		return nil
 	}
 
-	podSchedulingDirty := state.podSchedulingDirty
-	if len(schedulingCtx.Spec.PotentialNodes) == 0 {
+	if !state.preScored {
 		// PreScore was not called, probably because there was
 		// only one candidate. We need to ask whether that
```

Comment:
I don't understand this comment. We always call PreScore, unless the PreScore extension point is disabled.

Reply:
Are you sure? That doesn't match my experience. Let's see...

[`prioritizeNodes`](https://github.com/kubernetes/kubernetes/blob/9f3cfbbcdbc8971dddfd7a6ba2c8eff10b0c9c82/pkg/scheduler/schedule_one.go#L690) always calls `PreScore` but `prioritizeNodes` only gets called when there is more than one node:
https://github.com/kubernetes/kubernetes/blob/9f3cfbbcdbc8971dddfd7a6ba2c8eff10b0c9c82/pkg/scheduler/schedule_one.go#L409-L421

Reply:
ah, I get it now. I was getting confused with PreFilter.

Maybe you can say: "there was only one candidate that passed the Filters"

Reply:
Comment updated.